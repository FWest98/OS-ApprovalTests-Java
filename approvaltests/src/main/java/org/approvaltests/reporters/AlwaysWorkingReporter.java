package org.approvaltests.reporters;

import org.approvaltests.core.ApprovalFailureReporter;

public class AlwaysWorkingReporter implements EnvironmentAwareReporter
{
  private ApprovalFailureReporter wrapped;
  public AlwaysWorkingReporter(ApprovalFailureReporter wrapped)
  {
    this.wrapped = wrapped;
  }
  @Override
  public void report(String received, String approved)
  {
    getWrapped().report(received, approved);
  }
  @Override
  public boolean isWorkingInThisEnvironment(String forFile)
  {
    return true;
  }
  public ApprovalFailureReporter getWrapped()
  {
    return wrapped;
  }
}
