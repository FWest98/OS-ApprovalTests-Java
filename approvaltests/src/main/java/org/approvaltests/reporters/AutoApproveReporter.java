package org.approvaltests.reporters;

import java.io.File;

import org.approvaltests.core.VerifyResult;

import com.spun.util.io.FileUtils;

public class AutoApproveReporter implements ReporterWithApprovalPower
{
  @Override
  public void report(String received, String approved)
  {
    File a = new File(approved);
    if (a.exists())
    {
      a.delete();
    }
    File r = new File(received);
    FileUtils.copyFile(r, a);
  }
  @Override
  public VerifyResult approveWhenReported()
  {
    return VerifyResult.SUCCESS;
  }
}
