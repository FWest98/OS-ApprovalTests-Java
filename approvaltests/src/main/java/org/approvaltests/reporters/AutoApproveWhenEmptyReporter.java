package org.approvaltests.reporters;

import java.io.File;

import org.approvaltests.Approvals;
import org.approvaltests.core.ApprovalFailureReporter;
import org.approvaltests.core.VerifyResult;

import com.spun.util.io.FileUtils;

public class AutoApproveWhenEmptyReporter implements ReporterWithApprovalPower
{
  private final ApprovalFailureReporter reporter;
  private VerifyResult                  result;
  public AutoApproveWhenEmptyReporter()
  {
    this(Approvals.getReporter());
  }
  public AutoApproveWhenEmptyReporter(ApprovalFailureReporter reporter)
  {
    this.reporter = reporter;
  }
  @Override
  public void report(String received, String approved)
  {
    File a = new File(approved);
    if (!a.exists())
    {
      File r = new File(received);
      FileUtils.copyFile(r, a);
      result = VerifyResult.SUCCESS;
    }
    else
    {
      reporter.report(received, approved);
      result = VerifyResult.FAILURE;
    }
  }
  @Override
  public VerifyResult approveWhenReported()
  {
    return result;
  }
}
