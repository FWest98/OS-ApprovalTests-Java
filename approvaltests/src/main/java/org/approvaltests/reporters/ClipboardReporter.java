package org.approvaltests.reporters;

import com.spun.util.SystemUtils;
import com.spun.util.WindowUtils;

import java.awt.GraphicsEnvironment;
import java.io.File;

/**
 * A getReporter which creates the command to accept the received
 * file as the approve file and copies it to the clipboard:<br>
 * <code>move received.txt approved.txt</code>
 */
public class ClipboardReporter implements EnvironmentAwareReporter
{
  @Override
  public void report(String received, String approved)
  {
    WindowUtils.copyToClipBoard(getCommandLine(received, approved), false);
  }
  public static String getCommandLine(String received, String approved)
  {
    File r = new File(received);
    File a = new File(approved);
    String commandLine = ClipboardReporter.getAcceptApprovalText(r.getAbsolutePath(), a.getAbsolutePath());
    return commandLine;
  }
  /**
   * The clipboard will not be available in a headless environment.
   */
  @Override
  public boolean isWorkingInThisEnvironment(String forFile)
  {
    return !GraphicsEnvironment.isHeadless();
  }
  public static String getAcceptApprovalText(String received, String approved)
  {
    return getAcceptApprovalText(received, approved, SystemUtils.isWindowsEnviroment());
  }
  public static String getAcceptApprovalText(String received, String approved, boolean windowsEnviroment)
  {
    if (windowsEnviroment)
    {
      return String.format("move /Y \"%s\"  \"%s\"", received, approved);
    }
    else
    {
      return String.format("mv -f %s %s", formatFilePathForCommandLine(received),
          formatFilePathForCommandLine(approved));
    }
  }
  private static String formatFilePathForCommandLine(String filePath)
  {
    if (filePath.contains(" "))
    { return '"' + filePath + '"'; }
    return filePath;
  }
}
