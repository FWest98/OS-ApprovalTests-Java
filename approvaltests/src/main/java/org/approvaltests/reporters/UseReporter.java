package org.approvaltests.reporters;

import org.approvaltests.core.ApprovalFailureReporter;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

@Retention(RetentionPolicy.RUNTIME)
public @interface UseReporter {
  Class<? extends ApprovalFailureReporter>[] value();
}
