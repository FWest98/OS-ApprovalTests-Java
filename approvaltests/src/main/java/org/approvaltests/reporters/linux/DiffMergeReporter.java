package org.approvaltests.reporters.linux;

import org.approvaltests.reporters.DiffPrograms.Linux;
import org.approvaltests.reporters.GenericDiffReporter;

public class DiffMergeReporter extends GenericDiffReporter
{
  public static final DiffMergeReporter INSTANCE = new DiffMergeReporter();
  public DiffMergeReporter()
  {
    super(Linux.DIFF_MERGE);
  }
}