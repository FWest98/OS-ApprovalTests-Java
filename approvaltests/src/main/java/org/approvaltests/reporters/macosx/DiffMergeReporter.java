package org.approvaltests.reporters.macosx;

import org.approvaltests.reporters.DiffPrograms.Mac;
import org.approvaltests.reporters.GenericDiffReporter;

public class DiffMergeReporter extends GenericDiffReporter
{
  public static final DiffMergeReporter INSTANCE = new DiffMergeReporter();
  public DiffMergeReporter()
  {
    super(Mac.DIFF_MERGE);
  }
}
