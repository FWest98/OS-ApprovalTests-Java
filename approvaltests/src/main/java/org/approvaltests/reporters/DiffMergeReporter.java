package org.approvaltests.reporters;

import org.approvaltests.reporters.linux.DiffMergeLinuxReporter;
import org.approvaltests.reporters.macosx.DiffMergeMacOsReporter;

public class DiffMergeReporter extends FirstWorkingReporter
{
  public static final DiffMergeReporter INSTANCE = new DiffMergeReporter();
  public DiffMergeReporter()
  {
    super(DiffMergeMacOsReporter.INSTANCE, DiffMergeLinuxReporter.INSTANCE);
  }
}
