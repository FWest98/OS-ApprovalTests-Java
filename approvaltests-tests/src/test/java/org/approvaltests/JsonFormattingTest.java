package org.approvaltests;

import com.google.gson.GsonBuilder;
import org.approvaltests.reporters.UseReporter;
import org.approvaltests.reporters.macosx.DiffMergeReporter;
import org.junit.jupiter.api.Test;

@UseReporter(DiffMergeReporter.class)
public class JsonFormattingTest
{
  @Test
  public void testBasicFormatting()
  {
    String json = "{\"infos\":{\"address\":\"my address\",\"phone\":\"my phone\"},\"insurance\":{\"forks\":[14,53,123],\"prices\":[5,8,\"3%\"]}}";
    Approvals.verifyJson(json);
  }
  @Test
  public void testIncorrectFormatting()
  {
    String json = "{\"infos\":{address:my address,\"phone\":\"my phone\"},\"insurance\":{\"forks\":[14,53,123],\"prices\":[5,8,\"3%\"]}}";
    Approvals.verifyJson(json);
  }
  @Test
  void demonstrateNullIssueInVerifyJson()
  {
    Person person = new Person("Max", null, 1);
    Approvals.verifyAsJson(person, GsonBuilder::serializeNulls, GsonBuilder.class);
  }
  static class Person
  {
    public String firstName;
    public String lastName;
    public int    age;
    public Person(String firstName, String lastName, int age)
    {
      this.firstName = firstName;
      this.lastName = lastName;
      this.age = age;
    }
  }
}
