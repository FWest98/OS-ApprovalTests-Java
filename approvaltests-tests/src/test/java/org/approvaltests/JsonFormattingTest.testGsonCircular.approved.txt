java.lang.RuntimeException: Circular reference found.
Gson does not handle circular references.
Consider:
  1. Using XStream (JsonXstreamApprovals)
  2. Remove the circular reference.